app/commands/v2/put_content.rb

```diff
@@ -178,19 +170,23 @@ def lock_version_number_for_new_draft
 
       def user_facing_version_number_for_new_draft
         if previously_published_item != ITEM_NOT_FOUND
-          user_facing_version = UserFacingVersion.find_by!(content_item: previously_published_item)
-          user_facing_version.number + 1
+          previously_published_item.user_facing_version + 1
         else
           1
         end
       end
 
+      def pessimistic_content_item_scope
+        ContentItem.where(content_id: content_id, locale: locale).lock
+      end
+
       def previously_published_item
-        @previously_published_item ||= (
-          filter = ContentItemFilter.new(scope: pessimistic_content_item_scope)
-          content_items = filter.filter(state: %w(published unpublished), locale: locale)
-          UserFacingVersion.latest(content_items)
-        ) || ITEM_NOT_FOUND
+        @previously_published_item ||=
+          ContentItem.where(
+            content_id: content_id,
+            state: %w(published unpublished),
+            locale: locale,
+          ).order(user_facing_version: :desc).first || ITEM_NOT_FOUND
```

Comment:
Rather than using a `where` and `order` this should be a `find_by`. We shouldn't worry about two instances since we don't allow it a validation level